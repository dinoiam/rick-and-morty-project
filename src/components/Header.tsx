import React from 'react';
import styled from 'styled-components';
import { colors } from '@root/styles/colors';
import HeroSvg from '@root/images/hero.svg';

const HeaderComponent = styled.section`
  height: 200px;
  background-color: ${colors.background_variant};
  position: relative;
  padding: 20px;
  text-align: center;
  z-index: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
`;

const HeaderTitle = styled.span`
  color: ${colors.black};
  font-size: 50px;
`;

const Hero = styled(HeroSvg)`
  fill: ${colors.primary_variant};
  height: inherit;
  position: absolute;
  top: 0;
  left: 50%;
  transform: translateX(-50%);
  z-index: -1;
`;

export default function Header(): JSX.Element {
  return (
    <HeaderComponent>
      <HeaderTitle>THE RICK AND MORTY</HeaderTitle>
      <Hero></Hero>
    </HeaderComponent>
  );
}
