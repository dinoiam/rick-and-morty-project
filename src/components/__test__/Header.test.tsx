import React from 'react';
import Header from '@components/Header';
import { render } from '@testing-library/react';

describe('Header', () => {
  test('it renders properly', () => {
    const { container } = render(<Header></Header>);
    expect(container).toMatchSnapshot();
  });
});
