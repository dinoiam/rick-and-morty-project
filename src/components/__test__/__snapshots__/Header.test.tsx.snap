// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header it renders properly 1`] = `
<div>
  <section
    class="sc-bdfBwQ bfqCZ"
    data-testid="header-component"
  >
    <span
      class="sc-gsTCUz kxtmyK"
    >
      THE RICK AND MORTY
    </span>
    <div
      class="sc-dlfnbm jegWKw"
    />
  </section>
</div>
`;
